src/fp.rs

```diff
@@ -832,6 +837,67 @@ impl PrimeFieldBits for Fp {
     }
 }
 
+impl SerdeObject for Fp {
+    // Don't call this method of untrusted input. No checks performed before unsafe block.
+    fn from_raw_bytes_unchecked(bytes: &[u8]) -> Self {
+        debug_assert_eq!(bytes.len(), SIZE);
+        let mut out = blst_fp::default();
+        unsafe { blst_fp_from_lendian(&mut out, bytes.as_ptr()) };
+        Self(out)
+    }
+
+    fn from_raw_bytes(bytes: &[u8]) -> Option<Self> {
+        let input: [u8; SIZE] = bytes.try_into().expect(&format!("Expected {} bytes", SIZE));
+        Self::from_bytes_le(&input).into()
+    }
+
+    fn to_raw_bytes(&self) -> Vec<u8> {
+        let mut res = Vec::with_capacity(NUM_LIMBS * 8);
+        for limb in self.0.l.iter() {
+            res.extend_from_slice(&limb.to_le_bytes());
+        }
+        res
+    }
+
+    // Don't call this method of untrusted input. No checks performed before unsafe block.
+    fn read_raw_unchecked<R: std::io::Read>(reader: &mut R) -> Self {
+        let inner = [(); NUM_LIMBS].map(|_| {
+            let mut buf = [0; 8];
+            reader.read_exact(&mut buf).unwrap();
+            u64::from_le_bytes(buf)
+        });
+
+        let mut out = blst_fp::default();
+        unsafe { blst_fp_from_uint64(&mut out, inner.as_ptr()) };
+        Self(out)
+    }
+
+    fn read_raw<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let mut inner = [0u64; NUM_LIMBS];
+        for limb in inner.iter_mut() {
+            let mut buf = [0; 8];
+            reader.read_exact(&mut buf)?;
+            *limb = u64::from_le_bytes(buf);
+        }
```

Comment:
I prefer this notation than the one used in the function above. Could you use the same in `read_raw_unchecked`?